R/zchunk_L131.nonco2_proc_R_S_T_Y.R

```diff
@@ -36,55 +39,168 @@ module_emissions_L131.nonco2_proc_R_S_T_Y_DISABLED <- function(command, ...) {
+                                      "metals","foams","solvents", "semiconductors")) %>%
+       repeat_add_columns(tibble::tibble(Non.CO2 = c("CH4","N2O","NMVOC","NOx","SO2","CO","VOC"))) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     # Third: Group by and then join by sector, subsector, and Non.CO2.  However, to join the columns they have to have
+     # the same name. Rename column to tech_emissions. Tech_emissions is the sum and tech_share is the tech_emissions/sector_emissions
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, EPA_agg_fuel_ghg, Non.CO2) %>%
+       left_join(L131.EPA_nonco2_indproc.mlt, by = c(EPA_agg_sector = "sector", EPA_agg_fuel_ghg = "subsector","Non.CO2")) %>%
+       rename(tech_emissions = x) %>%
+       mutate(tech_emissions = if_else(is.na(tech_emissions),1,tech_emissions)) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, Non.CO2) %>% # group by sector, and Non CO2
+       summarise(tech_emissions = sum(tech_emissions)) ->
+       L131.nonco2_gg_R_prc_S_2005
+
+
+     L131.nonco2_pct_R_prc_S_S_2005 %>%
+       group_by(EPA_agg_sector, Non.CO2) %>%
+       left_join(L131.nonco2_gg_R_prc_S_2005, by = c("EPA_agg_sector", "Non.CO2")) %>%
+       rename(sector_emissions = tech_emissions.y ) %>% # sector emisisons from thr join
+       rename(tech_emissions = tech_emissions.x) %>%
+       mutate(tech_share = tech_emissions/sector_emissions) ->
+       L131.nonco2_pct_R_prc_S_S_2005
+
+     # Fourth: Disaggregate EDGAR emissions to subsectors
+     # First aggregate all EDGAR data and subset for processing sectors
+
+     #filter out years 2009 and 2010 in EDGAR_VOC
+     EDGAR_VOC %>%
+       filter(year!= "2009" & year != "2010") ->
+       EDGAR_VOC
+
+     # combine all of the nonco2s together in one file
+       bind_rows(EDGAR_CH4, EDGAR_N2O, EDGAR_VOC, EDGAR_NOx, EDGAR_SO2, EDGAR_CO, EDGAR_NH3) %>%
+       left_join(EDGAR_sector, by = c("IPCC_description", "IPCC")) %>%
+       rename(EDGAR_agg_sector = agg_sector) %>%
+       left_join(distinct(EDGAR_nation), by = "ISO_A3") %>%
+       left_join(iso_GCAM_regID, by = "iso") %>%
+       filter(EDGAR_agg_sector %in% c("industry_processes", "chemicals", "landfills", "wastewater","aerosols",
+                                "metals","foams","solvents", "semiconductors")) %>%
+       select(year,value, Non.CO2, EDGAR_agg_sector, GCAM_region_ID) %>%
+       na.omit()  %>%  # delete rows with NA's
+       group_by(year, Non.CO2, EDGAR_agg_sector, GCAM_region_ID) %>%
+       summarise(EDGAR_emissions = sum(value)) %>%
+       mutate(EDGAR_emissions = EDGAR_emissions * CONV_GG_TG) -> # convert from gg to tg
+       L131.EDGAR.mlt
+
+     L131.EDGAR.mlt$GCAM_region_ID <- as.integer(L131.EDGAR.mlt$GCAM_region_ID)
+     L131.EDGAR.mlt$year <- as.integer(L131.EDGAR.mlt$year)
+
+     # Fifth: map in all data and compute emissions (EDGAR_emissions * tech_share)
```

Comment:
This whole section (below) needs more comments to help reader understand what's going on.